src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -31454,6 +31775,26 @@ static void M(C c, Base b)
                 );
         }
 
+        [Fact]
+        public void NotNull_Indexer_WithAssignment()
+        {
+            var source =
+@"using System.Diagnostics.CodeAnalysis;
+public class C
+{
+    [NotNull] string? this[int i] { get => throw null!; set => throw null!; }
+    void M()
+    {
+        this[0].ToString();
+        this[0] = null;
```

Comment:
this behavior was the same before the PR because we didn't track a flow state for the indexer access, right?

Reply:
Correct :-)

---
In reply to: [324902235](https://github.com/dotnet/roslyn/pull/38610#discussion_r324902235) [](ancestors = 324902235)